x-pack/filebeat/module/kubernetes/coredns/config/coredns.yml

```diff
@@ -0,0 +1,14 @@
+type: log
+paths:
+{{ range $i, $path := .paths }}
+  - {{$path}}
+{{ end }}
+tags: {{.tags}}
+json.keys_under_root: true
+processors:
+  - dissect:
+      tokenizer: "%{timestamp} [%{log_level}] %{client}:%{port} - %{id} \"%{qtype} %{qclass} %{qname} %{proto} %{size} %{dnssec_ok} %{bufsize}\" %{rcode} %{rflags} %{rsize} %{duration}s"
+      field: "message"
+      target_prefix: "coredns"
+  - drop_fields: 
+      fields: [time]
```

Comment:
It'd be better to place all processing in the ingest pipeline.

Reply:
[Just like for Zeek](https://github.com/elastic/beats/pull/10916#discussion_r259965402), I'm carrying the fight over here ;-)

I like the idea of splitting out the logs here. It lets people use Beat processors that are inherently agent-side. There's no PID in this specific log, but one may want to do reverse DNS on the IPs.

I would directly name the fields following ECS here.

`"%{timestamp} [%{event.level}] %{client.ip}:%{client.port} ...`

And make sure to copy client.* to source.*. We're trying to always have `source.*` as a baseline in ECS.